opcua/common/subscription.py

```diff
@@ -208,39 +208,58 @@ def _get_node(self, nodeid):
+        if type(eventtypes) is not list:
+            evtypes.append(eventtypes)
+        else:
+            evtypes = eventtypes
+
+        evtype_node_list = []
+        for evtype in evtypes:
+            evtype_node_list.append(self._get_node(evtype))  # make sure we have a list of Node objects
+
         evfilter = ua.EventFilter()
-        evfilter.SelectClauses = self._select_clauses_from_evtype(eventtype)
-        evfilter.WhereClause = self._where_clause_from_evtype(eventtype)
+        evfilter.SelectClauses = self._select_clauses_from_evtype(evtype_node_list)
+        evfilter.WhereClause = self._where_clause_from_evtype(evtype_node_list)
         return evfilter
 
-    def _select_clauses_from_evtype(self, evtype):
+    def _select_clauses_from_evtype(self, evtypes):
         clauses = []
-        for prop in get_event_properties_from_type_node(evtype):
-            op = ua.SimpleAttributeOperand()
-            op.TypeDefinitionId = evtype.nodeid
-            op.AttributeId = ua.AttributeIds.Value
-            op.BrowsePath = [prop.get_browse_name()]
-            clauses.append(op)
+
+        selected_paths = []
+        for evtype in evtypes:
+            for prop in get_event_properties_from_type_node(evtype):
+                if prop.get_browse_name() not in selected_paths:
+                    op = ua.SimpleAttributeOperand()
+                    op.TypeDefinitionId = evtype.nodeid
+                    op.AttributeId = ua.AttributeIds.Value
+                    op.BrowsePath = [prop.get_browse_name()]
+                    clauses.append(op)
+                    selected_paths.append(prop.get_browse_name())
         return clauses
 
-    def _where_clause_from_evtype(self, evtype):
+    def _where_clause_from_evtype(self, evtypes):
         cf = ua.ContentFilter()
         el = ua.ContentFilterElement()
-        # operands can be ElementOperand, LiteralOperand, AttributeOperand, SimpleAttribute
-        op = ua.SimpleAttributeOperand()
-        op.TypeDefinitionId = evtype.nodeid
-        op.BrowsePath.append(ua.QualifiedName("EventType", 0))
-        op.AttributeId = ua.AttributeIds.Value
-        el.FilterOperands.append(op)
-        for subtypeid in [st.nodeid for st in self._get_subtypes(evtype)]:
-            op = ua.LiteralOperand()
-            op.Value = ua.Variant(subtypeid)
+
+        for evtype in evtypes:
+            # operands can be ElementOperand, LiteralOperand, AttributeOperand, SimpleAttribute
+            op = ua.SimpleAttributeOperand()
+            op.TypeDefinitionId = evtype.nodeid
+            op.BrowsePath.append(ua.QualifiedName("EventType", 0))
+            op.AttributeId = ua.AttributeIds.Value
             el.FilterOperands.append(op)
-        el.FilterOperator = ua.FilterOperator.InList
+            for subtypeid in [st.nodeid for st in self._get_subtypes(evtype)]:
+                op = ua.LiteralOperand()
+                op.Value = ua.Variant(subtypeid)
+                el.FilterOperands.append(op)
```

Comment:
I need to look at this closer because I don't really understand what you mean. I actually didn't read the spec on WhereClause yet.